### ISSUE TITLE
Strict causal padding

### ISSUE THREAD

**Issue by rueberger** [fchollet/keras#7202]
Currently, conv1D causal padding 0 pads. This can be undesirable, for instance when training Wavenet like architectures that have massive receptive fields - where when training on mini-batches 0 padding massively biases training.  

How challenging would it be to fix this?

**Comment by rueberger** [comment]
I guess this is already accomplished by setting padding = 'valid'?

**Comment by lingerlyn** [comment]
It's an edge effect, it only becomes an issue when the receptive field is comparable to your batch length. But if you're worried about it,  I think you can solve this issue by simply add a layer that discard the first several samples of receptive field.  Just do something like this:
out = Lambda(lambda a: a[:, receptive_field:, :])(out)
suppose your data is of shape [batch_size, time, dims].
